supabase/migrations/20240531142938_add_client_notes.sql

```diff
@@ -0,0 +1,104 @@
+alter table "public"."clients" add column "notes" text;
+
+set check_function_bodies = off;
+
+CREATE OR REPLACE FUNCTION public."insertClientAndTheirFamily"(clientrecord jsonb, familymembers jsonb)

```

Comment:
Hi @AmeliaIsCoding 
I know this was a while ago, but...

I'm unsure why the stored procedure in this migration is called `insertClientAndTheirFamily` - the rest of the PR looks like we should be changing the existing procedure `insert_client_and_family` (which is called from submitFormHelpers.ts.

Reply:
I think what's happened is [this commit](https://github.com/NorwoodAndBrixtonFoodbank/nbf-website/commit/2f5c9f774b35ed756358e30b69e9665acb1762a1) was merged by @minalui after this PR was raised.

Could I please check my thoughts with you both?

I _think_ I should:
1. merge `dev` into this branch (so that `insert_client_and_family` has replaced `insertClientAndTheirFamily`)
1. change the Function name in this this sql file
1. apply the outstanding migrations to my local DB
  - I think I should be able to use `npx supbase db push` rather than rebuilding my db from scratch - what have you been doing locally?
1. regenerate the types file using `npm run db:local:generate_types` (because in this PR it has the camelCase function name)

Thanks

Reply:
Hi Lincoln,

Sorry if I'm a little rusty - it has been a little while. @AmeliaIsCoding please feel free to correct if I've made a mistake.
I think I renamed it to insert_client_and_family to match the supabase syntax when I was refactoring the function to add birth year instead of age. It was one of my last tickets so my thought is it maybe the change wasn't merged yet?

I think your method should work - depending on the date of the migration file, you may need to change it to a later date. 

What I did previously to make changes to db:
1. Switch to `dev` branch
2. `npm run dev:reset_supabase`
3. Switch to current branch
4. Make changes needed in supabase (I think Amelia has added the client notes into the function)
5. `npx supabase db diff -f <name_of_migration>` to migrate the changes
6. `npm run db:local:generate_types` to generate the types
7. Then check if the migration can happen succesfully

Reply:
Thanks Mina :)